apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -231,46 +947,67 @@ tx_location(error, TxHash, C) ->
                 lager:debug("tx in Block ~p", [BlockHash]),
                 BlockHash
         end,
-    {L, C#{ {location, TxHash} => L }}.
+    {L, update_tx_log(TxHash, L, C)}.
 
-determine_depth(BHash, MinDepth, C) ->
-    {H, C1} = height(BHash, C),
-    case H of
-        undefined ->
-            {undefined, C1};
-        _ ->
-            determine_depth_(H, MinDepth, C1)
-    end.
+in_main_chain(BHash, C) ->
+    %% We don't want to use aec_chain_state:hash_is_in_main_chain/1, since we want to
+    %% stick to the top hash of the triggering event.
+    cached_get({in_main_chain, BHash}, C, fun(C1) -> in_main_chain_(BHash, C1) end).
+
+in_main_chain_(Hash, C) ->
+    {TopHash, C1} = top_hash(C),
+    Res = aec_chain_state:hash_is_in_main_chain(Hash, TopHash),
+    {Res, C1}.
+
+update_tx_log(TxHash, BlockHash, #{tx_log := TxLog} =C)
+  when is_binary(BlockHash) ->
+    C#{tx_log => aesc_window:add(
+                   { {TxHash, BlockHash},
+                     #{ tx_hash      => TxHash
+                      , block_hash   => BlockHash
+                      , block_origin => chain } }, TxLog)};
```

Comment:
The second element in this tuple that is the entry of the `aesc_window` data structure has no `type` key here.